rwengine/src/audio/Sound.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef _RWENGINE_SOUND_HPP_
+#define _RWENGINE_SOUND_HPP_
+
+#include <memory>
+#include <vector>
+#include <string>
+
+#include <al.h>
+#include <alc.h>
+#include <glm/glm.hpp>
+
+#include <rw/types.hpp>
+#include <rw/filesystem.hpp>
+
+#include "audio/SoundBuffer.hpp"
+#include "audio/SoundSource.hpp"
+
+/// Wrapper for SoundBuffer and SoundSource.
+/// Each command connected
+/// with playment is passed to SoundBuffer
+struct Sound {
+    size_t id = 0;
+    bool isLoaded = false;
+
+    std::shared_ptr<SoundSource> source = nullptr;
+    std::unique_ptr<SoundBuffer> buffer = nullptr;
+
+    Sound() = default;
+
+    bool isPlaying() const {
+        return buffer->isPlaying();
+    }
+
+    bool isPaused() const {
+        return buffer->isPaused();
+    }
+    bool isStopped() const {
+        return buffer->isStopped();
+    }
+
+    void play()  {
+        buffer->play();
+    }
+
+    void pause() {
+        buffer->pause();
+    }
+
+    void stop() {
+        buffer->stop();
+    }
+
+    void setPosition(const glm::vec3 position) {
+        buffer->setPosition(position);
+    }
+
+    void setLooping(const bool  looping) {
+        buffer->setLooping(looping);
+    }
+
+    void setPitch(const float  pitch) {
```

Comment:
Is the const really necessary? Same remark goes to all const arguments in this pr.